py4syn/epics/HexapodeClass.py

```diff
@@ -0,0 +1,255 @@
+"""Hexapode Bora controller class
+
+Python class for Hexapode Bora controllers
+
+:platform: Unix
+:synopsis: Python class for Hexapode Bora from Symetrie controllers
+
+
+.. moduleauthors:: 	Douglas henrique C. de Araújo <douglas.araujo@lnls.br> 
+		   	Luciano Carneiro Guedes <luciano.guedes@lnls.br>
+
+"""
+import time
+from threading import Event
+
+from epics import PV,Device, ca
+from py4syn.epics.IScannable import IScannable
+from py4syn.epics.StandardDevice import StandardDevice
+from py4syn.epics.MotorClass import Motor
+
+
+class Hexapode(IScannable,StandardDevice):
+    def __init__(self, mnemonic, pvName, axis):
+
+        super().__init__(mnemonic)
+        self.hexapode = Device(pvName + ':', ('STATE#PANEL:SET','STATE#PANEL:GET','STATE#PANEL:BUTTON','MOVE#PARAM:CM', 'MOVE#PARAM:X', 'MOVE#PARAM:Y', 'MOVE#PARAM:Z', 'MOVE#PARAM:RX', 'MOVE#PARAM:RY', 'MOVE#PARAM:RZ' ,':POSUSER:X',':POSUSER:Y',':POSUSER:Z',':POSUSER:RX',':POSUSER:RY',':POSUSER:RZ', ':POSMACH:X',':POSMACH:Y',':POSMACH:Z',':POSMACH:RX',':POSMACH:RY',':POSMACH:RZ','CFG#CS:1','CFG#CS:2', 'STATE#POSVALID?', 'CFG#CS?:1', 'CFG#CS?:2', 'CFG#CS?:3', 'CFG#CS?:4','CFG#CS?:5','CFG#CS?:6','CFG#CS?:7','CFG#CS?:8','CFG#CS?:9','CFG#CS?:10','CFG#CS?:11','CFG#CS?:12','CFG#CS?:13'))
+        self.axis=axis
+        self.axis_dic={"X":1,"Y":2,"Z":3,"RX":4,"RY":5,"RX":6}
+        self.axis_number=self.axis_dic[self.axis]
+        self.inicio1=0
+        self.rbv=PV(pvName + ':'+'POSUSER:'+self.axis)
+        self.pos=self.hexapode.get('POSUSER:'+self.axis)       
+        self.hexapode.add_callback('POSUSER:'+self.axis, self.onStatusChange)
+
+        self.hexapode.put('POSUSER:'+self.axis +".SCAN",9)
+
+    def onStatusChange(self,value,**kw):
+        """
+        Returns True if the target position was reachedy
+        """
+        if ( abs(float(value) - float(self.pos)) >1e-4):
```

Comment:
It is nice define constants at code begin. For 1e-4 you could do something like
MOVEERROR = 1e-4

This facilitate if you need to update this values in the future
(there are other constants like this on code)